File: AmiralBattiSwift/AmiralBattiSwift/models/JoinedRoomData.swift
//
//  JoinedRoomData.swift
//  AmiralBattiSwift
//
//  Created by Alperen Saraç on 10.04.2026.
//

import Foundation

struct JoinedRoomData: Codable {
    let roomCode: String
    let playerId: String
    let players: [PlayerInfo]
    let message: String
}

File: amiral_batti_flutter/lib/models/joined_room_data.dart
import 'package:amiral_batti_flutter/models/player_info.dart';

class JoinedRoomData {
  final String roomCode;
  final String playerId;
  final List<PlayerInfo> players;
  final String message;

  JoinedRoomData({
    required this.roomCode,
    required this.playerId,
    required this.players,
    required this.message,
  });

  factory JoinedRoomData.fromJson(Map<String, dynamic> json) {
    return JoinedRoomData(
      roomCode: json['roomCode'] ?? '',
      playerId: json['playerId'] ?? '',
      players: (json['players'] as List? ?? [])
          .map((e) => PlayerInfo.fromJson(e))
          .toList(),
      message: json['message'] ?? '',
    );
  }
}
